ZipBuilder/Sources/ZipBuilder/ShellUtils.swift

```diff
@@ -0,0 +1,171 @@
+/// xcodebuild, etc). Intentionally empty, this enum is used as a namespace.
+internal enum Shell {}
+
+extension Shell {
+  /// A type to represent the result of running a shell command.
+  enum Result {
+    /// The command was successfully run (based on the output code), with the output string as the
+    /// associated value.
+    case success(output: String)
+
+    /// The command failed with a given exit code and output.
+    case error(code: Int32, output: String)
+  }
+
+  /// Execute a command in the user's shell. This creates a temporary shell script and runs the
+  /// command from there, instead of calling via `Process()` directly in order to include the
+  /// appropriate environment variables. This is mostly for CocoaPods commands, but doesn't hurt
+  /// other commands.
+  ///
+  /// - Parameters:
+  ///   - command: The command to run in the shell.
+  ///   - outputToConsole: A flag if the command output should be written to the console as well.
+  ///   - workingDir: An optional working directory to run the shell command in.
+  /// - Returns: A Result containing output information from the command.
+  static func executeCommandFromScript(_ command: String,
+                                       outputToConsole: Bool = true,
+                                       workingDir: URL? = nil) -> Result {
+    let scriptPath: URL
+    do {
+      let tempScriptsDir = FileManager.default.temporaryDirectory(withName: "temp_scripts")
+      try FileManager.default.createDirectory(at: tempScriptsDir,
+                                              withIntermediateDirectories: true,
+                                              attributes: nil)
+      scriptPath = tempScriptsDir.appendingPathComponent("wrapper.sh")
+
+      // Write the temporary script contents to the script's path. CocoaPods complains when LANG
+      // isn't set in the environment, so explicitly set it here.
+      let contents = """
+      export PATH="/usr/local/bin:$PATH"
+      export LANG="en_US.UTF-8"
+      source ~/.bash_profile
+      \(command)
+      """
+      try contents.write(to: scriptPath, atomically: true, encoding: .utf8)
+    } catch let FileManager.FileError.failedToCreateDirectory(path, error) {
+      fatalError("Could not execute shell command: \(command) - could not create temporary " +
+        "script directory at \(path). \(error)")
+    } catch {
+      fatalError("Could not execute shell command: \(command) - unexpected error. \(error)")
+    }
+
+    // Remove the temporary script at the end of this function. If it fails, it's not a big deal
+    // since it will be over-written next time and won't affect the Zip file, so we can ignore
+    // any failure.
+    defer { try? FileManager.default.removeItem(at: scriptPath) }
+
+    // Let the process call directly into the temporary shell script we created.
+    let task = Process()
+    task.arguments = [scriptPath.path]
+
+    if #available(OSX 10.13, *) {
```

Comment:
Would an assert be better?  Do we realistically expect this to be run on 10.12 or less? It would seem better to not bother with the else code.

Reply:
I think it's fair to remove the else code and have an assert - I wanted to limit the entire package to only support 10.13+ but that feature isn't available until we drop Xcode 9 support (it's [implemented in Swift 5](https://github.com/apple/swift-evolution/blob/master/proposals/0236-package-manager-platform-deployment-settings.md))